temp/legacy-code/src/main/java/com/ivy/legacy/domain/pure/transaction/AccountValueFunctions.kt

```diff
@@ -52,7 +52,7 @@ object AccountValueFunctions {
         transaction: Transaction,
         accountId: UUID
     ): BigDecimal = with(transaction) {
-        if (this.getAccountId() == accountId && this is Transfer) {
+        if (this is Transfer && this.toAccount.value == accountId) {
```

Comment:
Shouldn't we do the same for `transferExpense`? Also can you please explain the logic behind this

Reply:
The logic is simple 

If the toAccountId of a transaction match with the current accountId in consideration then it is treated as income for that account 

For any Transaction 

X --> Y 

The amount will always be an income for Y account and expense for X account